plugins/flatpak/gs-flatpak.c

```diff
@@ -2499,10 +2499,23 @@ gs_flatpak_get_list_for_remove (GsFlatpak *self, GsApp *app,
 	for (guint i = 0; i < related->len; i++) {
 		FlatpakRelatedRef *xref_related = g_ptr_array_index (related, i);
 		g_autoptr(GsApp) app_tmp = NULL;
+		g_autoptr(GError) error_local = NULL;
+
 		if (!flatpak_related_ref_should_delete (xref_related))
 			continue;
 		app_tmp = gs_flatpak_create_app (self, FLATPAK_REF (xref_related));
 		gs_app_set_origin (app_tmp, gs_app_get_origin (app));
+		if (!gs_plugin_refine_item_state (self, app_tmp, cancellable, &error_local)) {
+			if (g_cancellable_is_cancelled (cancellable)) {
```

Comment:
Any reason not to use `g_error_matches()` instead? Sounds more robust.

Reply:
Well, I only really want to care about the cancellation. I could check the error for `G_IO_ERROR_CANCELLED` but then we have to assume that this is the one that gets assigned on cancellation. I will do it.